src/main/java/com/redhat/devtools/lsp4ij/client/features/EditorBehaviorFeature.java

```diff
@@ -176,9 +174,49 @@ public static boolean enableTextMateNestedBracesImprovements(@NotNull PsiFile fi
      * @return true if the semantic tokens-based file view provider is enabled; otherwise false
      */
     public static boolean enableSemanticTokensFileViewProvider(@NotNull PsiFile file) {
-        return LanguageServiceAccessor.getInstance(file.getProject()).hasAny(
-                file.getVirtualFile(),
-                ls -> ls.getClientFeatures().getEditorBehaviorFeature().isEnableSemanticTokensFileViewProvider(file)
+        return isEnabled(file, EditorBehaviorFeature::isEnableSemanticTokensFileViewProvider);
+    }
+
+    private interface FeatureFlagChecker {
+        boolean isEnabled(@NotNull EditorBehaviorFeature editorBehaviorFeature, @NotNull PsiFile file);
+    }
+
+    private static boolean isEnabled(
```

Comment:
Separated the signature for a `PsiFile` from the signature for a `Project`/`VirtualFile` pair to ensure that the original `PsiFile` isn't ever unintentionally "leaked" into the cache dependencies. This is quite important with `CachedValuesManager`.